tutorials/13-seasonal-time-series/13_seasonal_time_series.jmd

```diff
@@ -0,0 +1,328 @@
+This helps convergence while maintaining interpretability and the correct scales for the cyclic components.
+
+```julia
+σ_true = 0.35
+t = collect(tt[begin:3:end])
+t_min, t_max = extrema(t)
+x = (t .- t_min) ./ (t_max - t_min)
+yf = f.(t) .+ σ_true .* randn(size(t))
+yf_max = maximum(yf)
+yf = yf .- yf_max
+
+scatter(x, yf; title="Standardised data", legend=false)
+```
+
+Let us now build our model.
+We want to assume a linear trend, and cyclic effects.
+Encoding a linear trend is easy enough, but what about cyclical effects?
+We will take a scattergun approach, and create multiple cyclical features using both sine and cosine functions and let our inference machinery figure out which to keep.
+To do this, we define how long a one period should be, and create features in reference to said period.
+How long a period should be is problem dependent, but as an example let us say it is $1$ year.
+If we then find evidence for a cyclic effect with a frequency of 2, that would mean a biannual effect. A frequency of 4 would mean quarterly etc.
+Since we are using synthetic data, we are simply going to let the period be 1, which is the entire length of the time series.
+
+```julia
+freqs = 1:10
+num_freqs = length(freqs)
+period = 1
+cyclic_features = [sinpi.(2 .* freqs' .* x ./ period) cospi.(2 .* freqs' .* x ./ period)]
+
+plot_freqs = [1, 3, 5]
+freq_ptl = plot(
+    cyclic_features[:, plot_freqs];
+    label=permutedims(["sin(2π$(f)x)" for f in plot_freqs]),
+    title="Cyclical features subset",
+)
+```
+
+Having constructed the cyclical features, we can finally build our model. The model we will implement looks like this
+
+$$
+f(t) = \alpha + \beta_t t + \sum_{i=1}^F \beta_{\sin{},i} \sin{}(2\pi f_i t) + \sum_{i=1}^F \beta_{\cos{},i} \cos{}(2\pi f_i t),
+$$
+
+with a Gaussian likelihood $y \sim \mathcal{N}(f(t), \sigma^2)$.
+For convenience we are treating the cyclical feature weights $\beta_{\sin{},i} and \beta_{\cos{},i}$ the same in code and weight them with $\beta_c$.
+And just because it is so easy, we parameterise our model with the operation with which to apply the cyclic effects.
+This lets us use the exact same code for both additive and multiplicative models.
+Finally, we plot prior predictive samples to make sure our priors make sense.
+
+```julia
+@model function decomp_model(t, c, op)
+    α ~ Normal(0, 10)
+    βt ~ Normal(0, 2)
+    βc ~ MvNormal(zeros(size(c, 2)), I)
+    σ ~ truncated(Normal(0, 0.1); lower=0)
+
+    cyclic = c * βc
+    trend = α .+ βt .* t
+    μ = op(trend, cyclic)
+    y ~ MvNormal(μ, σ^2 * I)
+    return (; trend, cyclic)
+end
+
+y_prior_samples = mapreduce(hcat, 1:100) do _
+    rand(decomp_model(t, cyclic_features, +)).y
+end
+plot(t, y_prior_samples; linewidth=1, alpha=0.5, color=1, label="", title="Prior samples")
+scatter!(t, yf; color=2, label="Data")
+```
+
+With the model specified and with a reasonable prior we can now let Turing decompose the time series for us!
+
+```julia
```

Comment:
The HTML file is quite large, probably due to the size of some of the SVGs. I assume the problem might be this plot (and the similar one below). Maybe we can enforce png output here by setting chunk options? Eg.,

````suggestion
```julia; fig_ext = ".png"
````